### PR TITLE
ch4/am: move comm_abort am protocol into its own file

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Move handler_id enum table from `src/mpid/ch4/src/ch4_types.h` to `src/mpid/ch4/generic/am/mpidig_am.h`

Move `comm_abort` am protocol into its own file. The only exposed function is the protocol init function and one entrance function, all callbacks and active message header structure are kept internal.

Now that `comm_abort` protocol is self-contained, we can freely modify its `am_hdr` structure as fit.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
       reference the design discussed in #4236
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by hzhou** [comment]
test:jenkins/ch4/most

**Comment by hzhou** [comment]
test:jenkins/ch4/most

**Comment by hzhou** [comment]
test:mpich/warnings

**Comment by hzhou** [comment]
@yfguo Can you take a look at this PR?

File: src/mpid/ch4/generic/mpidig_am_comm_abort.c [hunk 1]
```diff
@@ -24,9 +24,7 @@ void MPIDIG_am_comm_abort_init(void)
 /* MPIDIG_COMM_ABORT */
 
 struct am_comm_abort_hdr {
-    int src_rank;
```
Comment (raffenet) [inline src/mpid/ch4/generic/mpidig_am_comm_abort.c]: This information would be helpful in an error or log message (rank %d called MPI_Abort on context_id %x). I suggest keeping it.

Comment (hzhou) [reply src/mpid/ch4/generic/mpidig_am_comm_abort.c]: The point of this PR is now that this structure is internal to this protocol, if you need any additional field, just add it and also add the corresponding code below in the sender and handler to use it. There is zero concern that you need a place holder for potential usage.

File: src/mpid/ch4/generic/mpidig.h [hunk 2]
```diff
@@ -15,6 +15,59 @@
...
+    MPIDIG_WIN_UNLOCK,
+    MPIDIG_WIN_UNLOCK_ACK,
+    MPIDIG_WIN_LOCKALL,
+    MPIDIG_WIN_LOCKALL_ACK,
+    MPIDIG_WIN_UNLOCKALL,
+    MPIDIG_WIN_UNLOCKALL_ACK,
+
+    MPIDIG_COMM_ABORT,
+
+    MPIDI_OFI_INTERNAL_HANDLER_CONTROL,
```
Comment (yfguo) [inline src/mpid/ch4/generic/mpidig.h]: Maybe we should think about not hard coding this one

**Comment by hzhou** [comment]
@gcongiu Since you are touching these files, would you like to review this PR?

**Comment by gcongiu** [comment]
@hzhou sure I will look at it.

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/warnings/quick

**Comment by hzhou** [comment]
test:jenkins/ch4/most

**Comment by hzhou** [comment]
Retest to confirm:

test:jenkins/ch4/ofi

**Pull request merged**